trusted/trts/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2022 The MobileCoin Foundation
+
+use mc_sgx_trts_sys::{sgx_is_outside_enclave, sgx_is_within_enclave};
+use std::ffi::c_void;
+use std::mem::size_of_val;
+
+/// Determine if the `memory` is outside of the enclave.
+/// # Arguments
+/// * `memory` - The memory to check for being fully outside of the enclave.
+///
+/// # Returns
+/// `true` if *all* of the memory is outside of the enclave.
+pub fn is_outside_enclave<T>(memory: &[T]) -> bool {
+    let size = size_of_val(memory);
+    let result = unsafe { sgx_is_outside_enclave(memory.as_ptr() as *const c_void, size as u64) };
+    match result {
+        1 => true,
+        0 => false,
+        _ => panic!("Unexpected result from SGX"),
```

Comment:
Let's log the unexpected value
```suggestion
        x => panic!("Unexpected result from sgx_is_outside_enclave: {}", x),
```